data-plane/src/acme/authorization.rs

```diff
@@ -0,0 +1,360 @@
+        type Value = Vec<Challenge<T>>;
+
+        fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+            formatter.write_str("a sequence of challenges")
+        }
+
+        fn visit_seq<S: serde::de::SeqAccess<'de>>(
+            self,
+            mut seq: S,
+        ) -> Result<Self::Value, S::Error> {
+            let mut challenges = Vec::new();
+            while let Some(challenge) = seq.next_element::<IntermediateChallenge>()? {
+                challenges.push(Challenge {
+                    account: None,
+                    r#type: challenge.r#type,
+                    url: challenge.url,
+                    status: challenge.status,
+                    validated: challenge.validated,
+                    error: challenge.error,
+                    token: challenge.token,
+                });
+            }
+            Ok(challenges)
+        }
+    }
+
+    deserializer.deserialize_seq(ChallengesVisitor(std::marker::PhantomData))
+}
+
+impl<T: AcmeClientInterface + Default> Order<T> {
+    pub fn get_account_and_directory(&self) -> Result<AccountAndDirectory<T>, AcmeError> {
+        let account = self
+            .account
+            .clone()
+            .ok_or(AcmeError::FieldNotFound("account".into()))?;
+        let directory = account
+            .directory
+            .clone()
+            .ok_or(AcmeError::FieldNotFound("directory".into()))?;
+
+        Ok((account, directory))
+    }
+
+    pub async fn authorizations(&self) -> Result<Vec<Authorization<T>>, AcmeError> {
+        let (account, directory) = self.get_account_and_directory()?;
+
+        let mut authorizations = vec![];
+
+        for authorization_url in self.authorization_urls.clone() {
+            let response = directory
+                .authenticated_request(
+                    &authorization_url,
+                    "POST",
+                    None,
+                    &account
+                        .private_key
+                        .clone()
+                        .ok_or(AcmeError::FieldNotFound("private_key".into()))?,
+                    &Some(account.id.clone()),
+                )
+                .await?;
+
+            let resp_bytes = hyper::body::to_bytes(response.into_body()).await?;
+            let body_str = from_utf8(&resp_bytes)?;
+
+            let mut authorization: Authorization<_> = serde_json::from_str(body_str)?;
+
+            authorization.account = Some(account.clone());
+            authorization.url = authorization_url;
+            for challenge in &mut authorization.challenges {
+                challenge.account = Some(account.clone())
+            }
+            authorizations.push(authorization)
+        }
+
+        Ok(authorizations)
+    }
+}
+
+type AccountAndDirectory<T> = (Arc<Account<T>>, Arc<Directory<T>>);
+
+impl<T: AcmeClientInterface + Default> Authorization<T> {
```

Comment:
Ooc why is there an additional constraint on `Default`?

Reply:
`serde_json::from_str()` requires it to deserialize any objects using the `AcmeClientInterface` type. Maybe there's a better way to specify that passing it all along? I wasn't able to see any way however

```
error[E0277]: the trait bound `T: Default` is not satisfied
    --> data-plane/src/acme/order.rs:249:35
     |
249  |         let mut order: Order<_> = serde_json::from_str(body_str)?;
     |                                   ^^^^^^^^^^^^^^^^^^^^ the trait `Default` is not implemented for `T`
```

Reply:
This was updated to be defined on the `AcmeClientInterface` definition